backend/users/views.py

```diff
@@ -1,3 +1,9 @@
-from django.shortcuts import render
+from .forms import CustomUserCreationForm
+from django.urls import reverse_lazy
+from django.views import generic
 
-# Create your views here.
+
+class SignUp(generic.CreateView):
+    form_class = CustomUserCreationForm
+    success_url = reverse_lazy('login')
+    template_name = 'registration/signup.html'
```

Comment:
I don't think this is needed as well. We need to move forward with as few changes as possible and when core badgeyay is ready, we'll make these changes. If someone can register and login via default django forms, it is good enough for now

Reply:
@iamareebjamal I understand that you want minimum changes but this is the most basic way of implementing signup in Django. I had to write this view to append the default UserCreationForm because we are using an AbstractUser, otherwise, it was giving too many errors. I didn't change back the AbstractUser to Default Django User model because you asked for minimum changes to the existing codebase.

Reply:
That is not true, Django itself recommends to use a custom user model and not the Default user model. https://docs.djangoproject.com/en/3.0/topics/auth/customizing/#auth-custom-user

Besides, it has all the URLs and forms necessary already https://docs.djangoproject.com/en/3.0/topics/auth/default/#module-django.contrib.auth.views

However, if you say so that it is required for you to do these changes, I'll take a look more closely and let you know

Reply:
@iamareebjamal Please look into it. I tried my best to do it within minimum changes. These are the [StackOverflow answers](https://stackoverflow.com/questions/54901493/manager-isnt-available-auth-user-has-been-swapped-for-members-customuser) I referred to, and I was getting `Attribute Error: Manager isn't available; 'auth.User' has been swapped for 'users.User'`

Reply:
Will check

Reply:
Thank you so much.